### PR TITLE
DEVEXP-492: Numbers webhook validate authentication header

### PR DESCRIPTION
**Pull request by JPPortier**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>com.sinch.sdk</groupId>
     <artifactId>sinch-sdk-java</artifactId>
-    <version>1.2.0-SNAPSHOT</version>
+    <version>1.2.1-SNAPSHOT</version>
```
Comment (asein-sinch) [inline pom.xml:7]: Should go to 1.3-next

File: core/src/main/com/sinch/sdk/core/utils/MapUtils.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class MapUtils {
+
+  /**
+   * @param headers Map to be transformed
+   * @return A cloned map with case-insensitive comparator when looking for key value
+   */
+  public static <T> Map<String, T> getCaseInsensitiveMap(Map<String, T> headers) {
+
+    // convert header keys to use case-insensitive map keys
+    Map<String, T> caseInsensitiveHeaders = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```
Comment (asein-sinch) [inline core/src/main/com/sinch/sdk/core/utils/MapUtils.java:15]: Super cool Java feature coming with static Comparator 🤩

File: client/src/test/java/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+    boolean authenticationResult =
+        webhooksService.validateAuthenticationHeader(secret, headerSet, body);
+
+    assertFalse(authenticationResult);
+  }
+
+  @Test
+  void checkValidateAuthenticatedNoSecret() {
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidationTest.java:70]: Can you also add a test to check a signature generated with an empty key?

File: client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java [hunk 4]
```diff
@@ -4,8 +4,19 @@
...
+import java.util.Map;
 
 public class WebHooksService implements com.sinch.sdk.domains.numbers.api.v1.WebHooksService {
+  private final NumbersWebhooksAuthenticationValidation authenticationChecker;
+
+  public WebHooksService(NumbersWebhooksAuthenticationValidation authenticationChecker) {
+    this.authenticationChecker = authenticationChecker;
+  }
+
+  public boolean validateAuthenticationHeader(
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java:17]: You can add the `@Override` tag to reference the addition to the com.sinch.sdk.domains.numbers.api.v1.WebHooksService interface on line 39

File: client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidation.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.Objects;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class NumbersWebhooksAuthenticationValidation {
+
+  public static final String SIGNATURE_HEADER = "X-Sinch-Signature";
+  static final String algorithm = "HmacSHA1";
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidation.java]:
```suggestion
  private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
```

File: pom.xml (hunk 1 above)
Comment (JPPortier) [reply pom.xml:7]: Plan was to follow these steps:
 - release it onto a 1.2.1 version for a dedicate Numbers improvement
 - then add it to a not yet planned 1.3-next

File: client/src/test/java/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidationTest.java (hunk 3 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidationTest.java:70]: Good catch  It highlight an exception not handled by function.
Thank you & Fixed.
Also added a new test against `webHooksService.validateAuthenticationHeader` function usage to ensure `WebhooksService` implements well validation call.

File: client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java (hunk 4 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java:17]: 👍 Done

File: client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidation.java (hunk 5 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/NumbersWebhooksAuthenticationValidation.java]: 🤦 Indeed !
FIXED 😉

File: client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java (hunk 4 above)
Comment (asein-sinch) [reply client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java:17]: I can't see it 🧐

Comment (asein-sinch) [reply client/src/main/com/sinch/sdk/domains/numbers/api/v1/adapters/WebHooksService.java:17]: I can see it 🤩

**Review by asein-sinch — APPROVED** [review]
👍
